autogen/README.md

```diff
@@ -0,0 +1,86 @@
+# Autogen Framework Model Interfaces
+The Autogen framework gives 3 major file based interfaces for models: `model.json`, `modelparameters.m`, and `sm_stop_verify.m` and 1 major interface available as a struct in the Matlab workspace: `mp`.
+
+## model.json
+`model.json` is a required configuration file shared between the Autogen Framework and the Autogen Tools. This covers both system-level properties as well as run-time tunable parameters for the model via registers.  
+`model.json` properties:  
+- `system`: An object to describe the overall system  
+    - `target`: Which hardware system is being targeted, the `audiomini` or `audioblade` (default: `audiomini`)  
+    - `sampleClockFrequency`: The sampling rate of the audio being processed (default: `48000`)  
+    - `systemClockFrequency`: Clock rate of the audio processing on the FPGA (default: `98304000`)  
+    - `inputDataType`: Object representing the data type of the audio input  
+        - `wordLength`: number of total bits in the data type  
+        - `fractionLength`: number of fractional bits in the fixed point number  
+        - `signed`: true if the data type is signed, false otherwise  
+    - `outputDataType`: Object reprenting the data type of the audio output
+    (Note: Currently unused, only inputDataType is read)
+        - `wordLength`: number of total bits in the data type  
+        - `fractionLength`: number of fractional bits in the fixed point number  
+        - `signed`: true if the data type is signed, false otherwise  
+    - `numberOfChannels`: number of channels in the audio to be processed (default: `2`)  
+- `devices`: An array of objects representing devices. Simulation support is only for one device.  
+    - device object  
+        - `name`: Name of the Simulink model  
+        - `registers`: An array of objects representing run-time tunable parameters as registers  
+        - register object  
+            - `dataType`: object representing the data type of the register  
+                - `fractionalBits`: number of fractional bits in the fixed point number  
+                - `signed`: true if the data type is signed, false otherwise  
+                - `type`: string representation of type ('boolean', 'string', 'int')  
+                - `totalBits`: number of total bits in the data type  
```

Comment:
I feel like we should decide between totalBits and fractionalBits, used with the register data type, versus wordLength and fractionLength

Reply:
Agreed. I don't have a strong preference for either, as I think I've used both... 

Do you have a preference? When I sketched out the new model.json config file, I was thinking that totalBits and fractionalBits were more clear than wordLength and fractionLength. However, I believe the fixed point toolbox uses wordLength and fractionLength; if that's so, we might as well follow the fixed point toolbox's convention.

Reply:
I have gone with the fixed point toobox's convention for now